scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+
+    private PrimaryHashMap<String, Channel> persistedChannels;
+
+    private Queue<Request> waitUntilQueue = new ArrayDeque<>();
+
+    private CompilerConfiguration compilerConfiguration;
+
+    private GroovyShell shell;
+
+    protected RecordManager recMan;
+
+    public AOSynchronization() {
+
+    }
+
+    public AOSynchronization(String statusFileDirectoryPath) {
+        compilerConfiguration = new CompilerConfiguration();
+        compilerConfiguration.setTargetBytecode(CompilerConfiguration.JDK8);
+        shell = new GroovyShell(this.getClass().getClassLoader(), compilerConfiguration);
+        this.statusFileDirectory = new File(statusFileDirectoryPath);
+        if (!statusFileDirectory.exists()) {
+            statusFileDirectory.mkdirs();
+        }
+        statusFile = new File(statusFileDirectory, DEFAULT_STORE_SESSION_NAME);
+        inMemoryChannels = new HashMap<>();
+
+    }
+
+    private boolean init(boolean firstAttempt) {
+        close();
+        try {
+            logger.info("Loading Record Manager from file : " + statusFile);
+            recMan = RecordManagerFactory.createRecordManager(statusFile.getCanonicalPath());
+            persistedChannels = recMan.hashMap(STATUS_RECORD_NAME);
+            logger.info("Content of persisted store : " + persistedChannels);
+            // This empty loop triggers InvalidClassException in case of serial version uid problems
+            for (Map.Entry<String, Channel> tempChannels : persistedChannels.entrySet())
+                ;
+            recMan.commit();
+            if (!firstAttempt) {
+                logger.info("Loading of job database successful after clean.");
+            }
+        } catch (Throwable e) {
+            if (firstAttempt) {
+                logger.error("Error occurred when loading channels database " + statusFile.getAbsolutePath() +
+                             ", now cleaning it and retrying.", e);
+                close();
+                recMan = null;
+                cleanDataBase();
+                init(false);
+            } else {
+                close();
+                throw new IllegalStateException("Error when loading database (even after cleaning it): " +
+                                                statusFile.getAbsolutePath(), e);
+            }
+        }
+        return true;
+    }
+
+    public boolean isStarted() {
+        logger.info("Store is started");
+        return true;
+    }
+
+    public boolean cleanDataBase() {
+
+        if (recMan != null) {
+            throw new IllegalStateException("Connection to a DB is established, cannot clean it");
+        }
+
+        logger.info("Cleaning database");
+
+        // delete all db files
+        File[] dbChannelFiles = statusFileDirectory.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                if (name.startsWith(DEFAULT_STORE_SESSION_NAME)) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this if-then-else statement by a single return statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1126)